src/azure-cli/azure/cli/command_modules/servicefabric/_validators.py

```diff
@@ -119,6 +119,13 @@ def validate_create_managed_cluster(cmd, namespace):
         if namespace.upgrade_mode == 'Manual':
             raise CLIError("--upgrade-cadence should only be used whe --upgrade-mode is set to 'Automatic'.")
 
+def validate_add_network_security_rule(cmd, namespace):
+    client = servicefabric_managed_client_factory(cmd.cli_ctx)
+    cluster = _safe_get_resource(client.managed_clusters.get,
+                                 (namespace.resource_group_name, namespace.cluster_name))
+
+    if cluster.cluster_state != 'Ready':
```

Comment:
cluster can be None, should account for this

Reply:
updated